pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -326,7 +326,27 @@ func (a *sqlInstanceAdapter) Delete(ctx context.Context) (bool, error) {
 }
 
 func (a *sqlInstanceAdapter) Export(ctx context.Context) (*unstructured.Unstructured, error) {
-	return nil, nil
+	if a.actual == nil {
+		return nil, fmt.Errorf("SQLInstance %q not found", a.resourceID)
+	}
+
+	sqlInstance, err := SQLInstanceGCPToKRM(a.actual)
+	if err != nil {
+		return nil, fmt.Errorf("error converting SQL Instance from API %w", err)
```

Comment:
Nit: Usually I'd prefer we use `SQLInstance` (the kind name) as a special term and don't separate it using white spaces as `SQL Instance` can be a much broader term. If it is to represent the GCP resource, I might use the GCP brand name (e.g. `Cloud SQL instance`) instead. But we don't have an agreed error message convention, so it's up to you to decide how to describe it.